barretenberg_wrapper/build.rs

```diff
@@ -215,18 +210,18 @@ fn main() {
         .expect("Couldn't write bindings");
 }
 
-fn link_lib_omp() {
-    //
-    // we are using clang, so we need to tell the linker where
-    // to search for lomp.
-    //
-    // For macOS, we do not need to supply the search path
-    // however for linux we do
-    if let OS::Linux = select_os() {
-        let llvm_dir = find_llvm_linux_path();
-        println!("cargo:rustc-link-search={}/lib", llvm_dir);
+fn link_lib_omp(toolchain: &'static str) {
+    // We are using clang, so we need to tell the linker where to search for lomp
+    match toolchain {
+        INTEL_LINUX | ARM_LINUX => {
+            let llvm_dir = find_llvm_linux_path();
+            println!("cargo:rustc-link-search={}/lib", llvm_dir)
+        }
+        INTEL_APPLE => println!("cargo:rustc-link-search=/usr/local/lib"),
+        ARM_APPLE => println!("cargo:rustc-link-search=/opt/homebrew/lib"),
```

Comment:
This fixes the error from `clang` on macOS failing to find `lomp` when no search path is provided:
```
error: linking with `cc` failed: exit status: 1
  = note: ld: library not found for -lomp
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```